diff --git a/include/RequestManager.h b/include/RequestManager.h
index 9335054..0e734b2 100644
--- a/include/RequestManager.h
+++ b/include/RequestManager.h
@@ -256,7 +256,7 @@ private:
      *    @param object over which the action was applied
      *    @param id id of the object, -1 for Pool, -2 for no-id objects
      *              (allocate error, parse error)
-     *    @param rc returned error code (NULL to ignore)
+     *    @param rc returned error code (0 to ignore)
      *    @returns string for logging
      */
     static string action_error (const string& method,
diff --git a/src/rm/RequestManagerImagePersistent.cc b/src/rm/RequestManagerImagePersistent.cc
index 95d3075..1a52510 100644
--- a/src/rm/RequestManagerImagePersistent.cc
+++ b/src/rm/RequestManagerImagePersistent.cc
@@ -137,7 +137,7 @@ error_authorize:
 
 error_persistent:
     image->unlock();
-    oss << action_error(method_name, "MANAGE", "IMAGE", iid, NULL)
+    oss << action_error(method_name, "MANAGE", "IMAGE", iid, 0)
         << " Is the image public? An Image cannot be public and persistent.";
     goto error_common;
 
