diff --git a/src/tm_mad/nfs/tm_clone.sh b/src/tm_mad/nfs/tm_clone.sh
index 2bba596..15ea47a 100755
--- a/src/tm_mad/nfs/tm_clone.sh
+++ b/src/tm_mad/nfs/tm_clone.sh
@@ -39,14 +39,25 @@ log "DST: $DST_PATH"
 
 DST_DIR=`dirname $DST_PATH`
 
+
 log "Creating directory $DST_DIR"
 exec_and_log "mkdir -p $DST_DIR"
 exec_and_log "chmod a+w $DST_DIR"
 
 case $SRC in
-http://*)
+http://*.gz|https://*.gz)
+    log "Downloading GZip archive $SRC"
+    exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP"
+    ;;
+
+http://*.bz2|https://*.bz2)
+    log "Downloading BZip2 archive $SRC"
+    exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2"
+    ;;
+
+http://*|https://)
     log "Downloading $SRC"
-    exec_and_log "$WGET -O $DST_PATH $SRC"
+    exec_and_log "$CURL -k -o $DST_PATH $SRC"
     ;;
 
 *)
diff --git a/src/tm_mad/ssh/tm_clone.sh b/src/tm_mad/ssh/tm_clone.sh
index 9e6ff36..960c59f 100755
--- a/src/tm_mad/ssh/tm_clone.sh
+++ b/src/tm_mad/ssh/tm_clone.sh
@@ -43,9 +43,19 @@ log "Creating directory $DST_DIR"
 exec_and_log "$SSH $DST_HOST mkdir -p $DST_DIR"
 
 case $SRC in
-http://*)
+http://*.gz|https://*.gz)
+    log "Downloading GZip archive $SRC"
+    exec_and_log "download_and_extract $SRC $DST_PATH.gz $GUNZIP $DST_HOST"
+    ;;
+
+http://*.bz2|https://*.bz2)
+    log "Downloading BZip2 archive $SRC"
+    exec_and_log "download_and_extract $SRC $DST_PATH.bz2 $BUNZIP2 $DST_HOST"
+    ;;
+
+http://*|https://)
     log "Downloading $SRC"
-    exec_and_log "$SSH $DST_HOST $WGET -O $DST_PATH $SRC"
+    exec_and_log "$SSH $DST_HOST $CURL -k -o $DST_PATH $SRC"
     ;;
 
 *)
@@ -54,5 +64,5 @@ http://*)
     ;;
 esac
 
-exec_and_log "$SSH $DST_HOST chmod a+rw $DST_PATH"
+exec_and_log "$SSH $DST_HOST chmod a+w $DST_PATH"
 
diff --git a/src/tm_mad/tm_common.sh b/src/tm_mad/tm_common.sh
index 5caba58..758e847 100644
--- a/src/tm_mad/tm_common.sh
+++ b/src/tm_mad/tm_common.sh
@@ -39,6 +39,9 @@ SED=/bin/sed
 SSH=/usr/bin/ssh
 SUDO=/usr/bin/sudo
 WGET=/usr/bin/wget
+CURL=/usr/bin/curl
+GUNZIP=/bin/gunzip
+BUNZIP2=/bin/bunzip2
 
 function get_vmdir
 {
@@ -164,3 +167,18 @@ function timeout_exec_and_log
     fi
 }
 
+function download_and_extract
+{
+    URL=$1
+    ARCHIVE=$2
+    EXTRACT_CMD=$3
+    HOST=$4
+
+    if [ "x$HOST" != "x" ]; then
+        SSH_CMD="$SSH $HOST"
+    fi
+
+    $SSH_CMD $CURL -k -o $ARCHIVE $URL
+    $SSH_CMD $EXTRACT_CMD $ARCHIVE
+}
+
