diff --git a/src/tm_mad/nfs/tm_context.sh b/src/tm_mad/nfs/tm_context.sh
index b6c515d..c1c230c 100755
--- a/src/tm_mad/nfs/tm_context.sh
+++ b/src/tm_mad/nfs/tm_context.sh
@@ -31,10 +31,19 @@ done
 
 if [ -z "${ONE_LOCATION}" ]; then
     TMCOMMON=/usr/lib/one/mads/tm_common.sh
+    CONFIG_SCRIPT=/usr/share/one/scripts/configuration.sh
 else
     TMCOMMON=$ONE_LOCATION/lib/mads/tm_common.sh
+    CONFIG_SCRIPT=$ONE_LOCATION/share/scripts/configuration.sh
 fi
 
+if [ -z "${STRATUSLAB_LOCATION}" ]; then
+    STRATUS_CONFIG=/usr/bin/stratus-config
+else
+    STRATUS_CONFIG=$STRATUSLAB_LOCATION/scripts/stratus-config
+fi
+
+
 . $TMCOMMON
 
 get_vmdir
@@ -48,6 +57,12 @@ ISO_DIR=$DST_DIR/isofiles
 
 exec_and_log "mkdir -p $ISO_DIR"
 
+exec_and_log "cp $CONFIG_SCRIPT $ISO_DIR"
+
+CONTEXT_SCRIPT=$($STRATUS_CONFIG context_script)
+
+exec_and_log "cp $CONTEXT_SCRIPT $ISO_DIR"
+
 for f in $SRC; do
     case $f in
     http://*)
@@ -64,4 +79,3 @@ exec_and_log "$MKISOFS -o $DST_PATH -J -R $ISO_DIR"
 
 exec_and_log "rm -rf $ISO_DIR"
 
-
diff --git a/src/tm_mad/ssh/tm_context.sh b/src/tm_mad/ssh/tm_context.sh
index b420be3..2c9a8ad 100755
--- a/src/tm_mad/ssh/tm_context.sh
+++ b/src/tm_mad/ssh/tm_context.sh
@@ -31,10 +31,19 @@ done
 
 if [ -z "${ONE_LOCATION}" ]; then
     TMCOMMON=/usr/lib/one/mads/tm_common.sh
+    CONFIG_SCRIPT=/usr/share/one/scripts/configuration.sh
 else
     TMCOMMON=$ONE_LOCATION/lib/mads/tm_common.sh
+    CONFIG_SCRIPT=$ONE_LOCATION/share/scripts/configuration.sh
 fi
 
+if [ -z "${STRATUSLAB_LOCATION}" ]; then
+    STRATUS_CONFIG=/usr/bin/stratus-config
+else
+    STRATUS_CONFIG=$STRATUSLAB_LOCATION/scripts/stratus-config
+fi
+
+
 . $TMCOMMON
 
 
@@ -48,6 +57,12 @@ ISO_DIR="$TMP_DIR/isofiles"
 
 exec_and_log "mkdir -p $ISO_DIR"
 
+exec_and_log "cp $CONFIG_SCRIPT $ISO_DIR"
+
+CONTEXT_SCRIPT=$($STRATUS_CONFIG context_script)
+
+exec_and_log "cp $CONTEXT_SCRIPT $ISO_DIR"
+
 for f in $SRC; do
     case $f in
     http://*)
