diff --git a/src/vmm/LibVirtDriver.cc b/src/vmm/LibVirtDriver.cc
index c55468d..e579e10 100644
--- a/src/vmm/LibVirtDriver.cc
+++ b/src/vmm/LibVirtDriver.cc
@@ -143,7 +143,7 @@ int LibVirtDriver::deployment_description(
 
     if (emulator == "kvm")
     {
-        file << "\t\t<type>hvm</type>" << endl;
+        file << "\t\t<type machine='pc'>hvm</type>" << endl;
     }
 
     num = vm->get_template_attribute("OS",attrs);
