diff --git a/src/tm_mad/nfs/tm_clone.sh b/src/tm_mad/nfs/tm_clone.sh
index ff2ed67..603128a 100755
--- a/src/tm_mad/nfs/tm_clone.sh
+++ b/src/tm_mad/nfs/tm_clone.sh
@@ -44,9 +44,24 @@ exec_and_log "mkdir -p $DST_DIR"
 exec_and_log "chmod a+w $DST_DIR"
 
 case $SRC in
-http://*)
+http*.tar.gz|http*.tgz)
+    log "Downloading tar.gz archive $SRC"
+    exec_and_log "download_extract $SRC $DST_PATH.tar.gz $DST_PATH"
+    ;;
+
+http*.bz2)
+    log "Downloading tar.bz2 archive $SRC"
+    exec_and_log "download_extract $SRC $DST_PATH.tar.bz2 $DST_PATH"
+    ;;
+
+http*.xz)
+    log "Downloading tar.xz archive $SRC"
+    exec_and_log "download_extract $SRC $DST_PATH.tar.xz $DST_PATH"
+    ;;
+
+http*)
     log "Downloading $SRC"
-    exec_and_log "wget -O $DST_PATH $SRC"
+    exec_and_log "curl -k -o $DST_PATH $SRC"
     ;;
 
 *)
